romana/cmd/policy.go

```diff
@@ -0,0 +1,206 @@
+		return util.UsageError(cmd, "TENANT and POLICY FILE name should be provided.")
+	}
+
+	tenantName := args[0]
+	policyFile := args[1]
+	// Tenant check once adaptor add supports for it.
+	/*
+		if !adaptor.TenantExists(tnt) {
+			return errors.New("Tenant doesn't exists: " + tnt)
+		}
+	*/
+
+	f, err := os.Open(policyFile)
+	if err != nil {
+		return errors.New("Couldn't open Policy file: " + policyFile)
+	}
+	defer f.Close()
+
+	baseURL := config.GetString("BaseURL")
+	kubeURL := baseURL + fmt.Sprintf(":8080/apis/romana.io/demo/v1")
+	kubeURL = kubeURL + fmt.Sprintf("/namespaces/%s/networkpolicys", tenantName)
+
+	req, err := http.NewRequest("POST", kubeURL, f)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == 201 {
+		if config.GetString("Format") == "json" {
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return err
+			}
+			fmt.Printf(util.JSONIndent(string(body)))
+		} else {
+			fmt.Printf("Policy (%s) for Tenant (%s) successfully created.\n", policyFile, tenantName)
+		}
+		return nil
+	} else {
+		if config.GetString("Format") == "json" {
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return err
+			}
+			fmt.Printf(util.JSONIndent(string(body)))
+			return nil
+		} else {
+			return fmt.Errorf("Error creating Policy (%s) for Tenant (%s).", policyFile, tenantName)
+		}
+	}
+}
+
+// policyRemove removes kubernetes policy for a specific tenant
+// using the policyName provided.
+func policyRemove(cmd *cli.Command, args []string) error {
+	if len(args) < 2 {
+		return util.UsageError(cmd, "TENANT and POLICY name should be provided.")
+	}
+
+	tenantName := args[0]
+	policyNames := args[1:]
+	// Tenant check once adaptor add supports for it.
+	/*
+		if !adaptor.TenantExists(tnt) {
+			return errors.New("Tenant doesn't exists: " + tnt)
+		}
+	*/
+
+	var errs []error
+	baseURL := config.GetString("BaseURL")
+	kubeURL := baseURL + fmt.Sprintf(":8080/apis/romana.io/demo/v1")
+	for _, policyName := range policyNames {
+		k := kubeURL + fmt.Sprintf("/namespaces/%s/networkpolicys/%s/", tenantName, policyName)
+
+		req, err := http.NewRequest("DELETE", k, nil)
+		if err != nil {
+			return err
+		}
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == 200 {
+			if config.GetString("Format") == "json" {
+				body, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					return err
+				}
+				fmt.Printf(util.JSONIndent(string(body)))
+			} else {
+				fmt.Printf("Policy (%s) for Tenant (%s) successfully deleted.\n", policyName, tenantName)
+			}
+		} else {
+			if config.GetString("Format") == "json" {
+				body, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					return err
+				}
+				fmt.Printf(util.JSONIndent(string(body)))
```

Comment:
the json output there is pretty self explanatory, but i can add additional values here if wanted? the way the -f=json on command line works is that all output including errors are valid json and thus no extra printing or logging there to prevent clobbering the output which can be easily piped to utilities jq, etc

```
{
	"kind": "Status",
	"apiVersion": "v1",
	"metadata": {},
	"status": "Failure",
	"message": "thirdpartyresourcedatas.extensions \"pol2\" not found",
	"reason": "NotFound",
	"details": {
		"name": "pol2",
		"group": "extensions",
		"kind": "thirdpartyresourcedatas"
	},
	"code": 404
}
```